### PR TITLE
Add verify and join org link to verification email

### LINKED ISSUE #586: Add verify and join org link to verification email
When users create an account after trying to join an organization, we should keep track of that and make them automatically join the organization after verifying their email address.

Unlike when normally signing up, users should land on the organization homepage instead of the main districtbuilder homepage.

### PR DESCRIPTION
**Pull request by kevinearldenny**
## Overview

When a new user attempts to sign up for an account and join an organization from an organization's page, they are automatically added to the Organization as an 'unverified user'. If an unverified user views their organization's page before confirming their email, they are prompted to confirm their email.

### Checklist

- [x] Description of PR is in an appropriate section of `CHANGELOG.md` and grouped with similar changes, if possible

### Demo

image

### Notes

Optional. Ancillary topics, caveats, alternative strategies that didn't work out, anything else.

## Testing Instructions

- Ensure that you are not logged in with an existing user account
- Navigate to an organization page (e.g., http://localhost:3003/o/mikes-maps)
- Click 'Join Organization' and fill in credentials not associated with an existing user

Closes #586 & #609

### CODE REVIEW

File: src/client/screens/OrganizationScreen.tsx [hunk 1]
```diff
@@ -147,6 +151,49 @@ const OrganizationScreen = ({ organization, project, user }: StateProps) => {
...
+                Error resending email. If this error persists, please contact us at{" "}
+                <Styled.a sx={{ color: "muted" }} href="mailto:support@districtbuilder.org">
+                  support@districtbuilder.org
+                </Styled.a>
+                .
+              </Box>
+            )}
+          </Box>
+        </Alert>
+      )}
```
Comment (maurizi) [inline src/client/screens/OrganizationScreen.tsx:196]: Refactoring this banner was supposed to be tackled by https://github.com/PublicMapping/districtbuilder/issues/609 but in any case, we should just copy/paste the markup here - we're already sharing the `SiteHeader` component between the two pages, we should just _move_ [this code in `HomeScreen`](https://github.com/PublicMapping/districtbuilder/blob/develop/src/client/screens/HomeScreen.tsx#L78-L120) into `SiteHeader` as well and we won't need these changes.

Comment (kevinearldenny) [reply src/client/screens/OrganizationScreen.tsx:187]: Gotcha, okay - guess it probably makes sense to pull #609 into this PR as well

**Comment by kevinearldenny** [comment]
@maurizi Merge conflicts are now resolved here, ready for review

File: src/client/screens/OrganizationScreen.tsx [hunk 2]
```diff
@@ -223,26 +223,49 @@ const OrganizationScreen = ({ organization, user }: StateProps) => {
...
+                          : "You must confirm your email before joining an organization"}
+                      </div>
+                    }
                   >
                     {joinButton}
                   </Tooltip>
                 )
               ) : (
-                joinButton
+                <div></div>
```
Comment (maurizi) [inline src/client/screens/OrganizationScreen.tsx:238]: I don't think you meant to get rid of the Join button for logged out users?
```suggestion
                joinButton
```

File: src/client/screens/OrganizationScreen.tsx [hunk 3]
```diff
@@ -223,26 +223,49 @@ const OrganizationScreen = ({ organization, user }: StateProps) => {
...
+                            </div>
+                          }
+                        >
+                          <TemplateCard template={template} key={template.id} />
+                        </Tooltip>
+                      ) : (
+                        <TemplateCard template={template} key={template.id} />
+                      )
+                    )}
+                  </Box>
```
Comment (maurizi) [inline src/client/screens/OrganizationScreen.tsx:265]: This is not correct - the tooltip should be wrapped around the `<Button>`, not the `<TemplateCard>` component created in #615.

**Review by maurizi — CHANGES_REQUESTED** [review]

**Review by maurizi — CHANGES_REQUESTED** [review]
The main functionality of storing intent to join an organization as part of registering is working, and I was able to activate my account and was a part of the organization.

This is still missing the feature to redirect to the organization upon activation, as per the wireframes on Abstract:
image

As we discussed offline that will likely require some kind of new flag on `User` so we don't show that modal again, or thinking about it a bit more - storing a separate FK field on `User` for organization(s)-to-join instead of reusing the existing `organizations` M2M field could serve this purpose as well (I think you originally suggested this).

**Review by maurizi — APPROVED** [review]

**Pull request merged**